UnityProject/Assets/Scripts/Util/SweetExtensions.cs

```diff
@@ -171,6 +171,18 @@ public static Vector3 AssumedWorldPosServer(this GameObject go)
 		return GetRootGameObject(go).transform.position;
 	}
 
+
+	public static Matrix OnMatrixRoot(this GameObject go)
```

Comment:
the name of this method sounds like an event and doesn't immediately tell you you are going to get the matrix the object is sitting on. ``GetRootMatrix`` might be more appropriate.